### PR TITLE
Replace last instance of Contact::select

### PR DESCRIPTION
**Pull request by MrPetovan**
Follow-up to #7417

### CODE REVIEW

File: include/api.php [hunk 1]
```diff
@@ -6223,13 +6223,10 @@ function api_friendica_profile_show($type)
...
 			$users = [];
-			$nurls = Contact::select(['id', 'nurl'], ['uid' => api_user(), 'profile-id' => $rr['id']]);
-
-			if (DBA::isResult($nurls)) {
-				foreach ($nurls as $nurl) {
-					$user = api_get_user($a, $nurl['nurl']);
-					($type == "xml") ? $users[$k++ . ":user"] = $user : $users[] = $user;
-				}
+			$nurls = Contact::selectToArray(['id', 'nurl'], ['uid' => api_user(), 'profile-id' => $rr['id']]);
+			foreach ($nurls as $nurl) {
```
Comment (annando) [inline include/api.php]: The function ```Contact::selectToArray``` can return a boolean value when an SQL error occurred.

Comment (MrPetovan) [reply include/api.php]: Then it's a bad function and I will change it immediately.

Comment (MrPetovan) [reply include/api.php]: Alright, fixed. I found a couple more issues with Item and Attach.

Comment (annando) [reply include/api.php]: The boolean value indicates an error - or ```true``` when applied to some SQL query that doesn't return a dataset.

Comment (MrPetovan) [reply include/api.php]: Which isn't the case for `toArray` which expects a dataset.

File: src/Model/Item.php [hunk 2]
```diff
@@ -383,16 +383,10 @@ public static function selectToArray(array $fields = [], array $condition = [],
...
-
-		$data = [];
-		while ($row = self::fetch($result)) {
-			$data[] = $row;
+			return [];
 		}
-		DBA::close($result);
 
-                return $data;
+        return DBA::toArray($result);
```
Comment (annando) [inline src/Model/Item.php]: You mustn't use ```DBA::toArray(``` here. We have to rely upon ```Item::fetch(``` here, since that function is doing some magic here.

Comment (MrPetovan) [reply src/Model/Item.php]: Ha, I didn't notice the `$this->fetch()`.

File: src/Database/Database.php [hunk 3]
```diff
@@ -1512,16 +1512,18 @@ public function count($table, array $condition = [])
 	public function toArray($stmt, $do_close = true)
 	{
 		if (is_bool($stmt)) {
-			return $stmt;
```
Comment (annando) [inline src/Database/Database.php]: I would vote for moving this change into a separate PR, since it changes functionality and we have to make sure that it doesn't break anything which will cost some time - while the main issue is unfixed.

File: src/Model/Item.php (hunk 2 above)
Comment (annando) [reply src/Model/Item.php]: That's why I would like to have it into separate PRs :-)

File: src/Database/Database.php (hunk 3 above)
Comment (MrPetovan) [reply src/Database/Database.php]: On the contrary, we just have been lucky all this time. I checked  `DBA::toArray` usage and we systematically expect an array. This means no query have been erroring out, mostly because we use the `DBA::select` syntax as well that leaves little margin to make a SQL mistake.

File: src/Model/Item.php (hunk 2 above)
Comment (MrPetovan) [reply src/Model/Item.php]: Reverted the faulty change.

File: include/api.php (hunk 1 above)
Comment (annando) [reply include/api.php]: You could apply this function to a database command like "UPDATE" - then it would return a "true" as boolean.

Comment (MrPetovan) [reply include/api.php]: But it doesn't make sense functionally, there's no array to be extracted from an UPDATE query statement, so the empty array would be functionally correct.

Now, if you check the boolean state of the return value of `DBA::toArray` on an UPDATE statement to guess if the update was successful, you're doing all kinds of wrongs that have nothing to do with `DBA::toArray` anymore.

Comment (annando) [reply include/api.php]: I assumed that this function had been used in the legacy ```q(...)``` function - where we don't know if this is a "SELECT" or an "UPDATE". But luckily we do check for boolean return values before that function call.
